source/arm9/math.c

```diff
@@ -0,0 +1,38 @@
+
+// SPDX-License-Identifier: Zlib
+// Copyright (C) 2024 Dominik Kurz
+#include <math.h>
+
+
+float hw_sqrtf(float x)
+{
+    union{f32 f; u32 i;}xu;
+    xu.f = x;
+    u64 mantissa = xu.i & ((1 << 23) - 1); // extracts mantissa via bitmask
+    mantissa = (mantissa + (1 << 23)) << (((xu.i & (1<<23)) == 0) + 25); // applies additional shift depending on whether exponent is even or odd
+    REG_SQRTCNT = SQRT_64;  
+    REG_SQRT_PARAM = mantissa;
+    u32 raw_exponent = (xu.i & (0xff << 23));
+    // check if exponent is 0
+    if (raw_exponent > 0) 
+    {
```

Comment:
You're using tabs and spaces in the same function, replace everything by spaces. In here I don't see indentation correctly, for example. And people may have set up tabs to be 4 spaces, not 8 (which I think is your setup).